pkg/relayer/config/types.go

```diff
@@ -163,15 +168,31 @@ type RelayMinerMetricsConfig struct {
 type RelayMinerSupplierConfig struct {
 	// ServiceId is the serviceId corresponding to the current configuration.
 	ServiceId string
+
 	// ServerType is the transport protocol used by the supplier, it must match the
 	// type of the relay miner server it is associated with.
-
 	ServerType RelayMinerServerType
-	// ServiceConfig is the config of the service that relays will be proxied to.
-	// Other supplier types may embed other fields in the future. eg. "https" may
-	// embed a TLS config.
+
+	// [REQUIRED] ServiceConfig is the default config of the service that relays will be
```

Comment:
Per my comment above, have the TODO here and keep the optional/required details to the docs.